src/main/java/io/supertokens/passwordless/Passwordless.java

```diff
@@ -440,58 +440,84 @@ public static ConsumeCodeResponse consumeCode(TenantIdentifier tenantIdentifier,
             }
         }
 
+        if (user != null) {
```

Comment:
in this function, you need to only delete the code if deleteCode is true. Else, you need to do everything else you are doing, except for actually deleting the code